src/components/Dropzone.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+import React from 'react';
+import { Box, Button, Typography, useTheme } from '@mui/material';
+import { CloudUpload } from '@mui/icons-material';
+import type { ImageLoadEvent, DropzoneState } from '@/types';
+
+/**
+ * Props for the Dropzone component.
+ *
+ * @interface DropzoneProps
+ *
+ * @property {function} handleFileUpload - Function to handle file upload events.
+ * @property {function} handleDragOver - Function to handle drag over events.
+ * @property {function} handleDrop - Function to handle drop events.
+ * @property {function} handleImageLoad - Function to handle image load events.
+ * @property {DropzoneState} dropzoneState - State of the dropzone.
+ * @property {React.Dispatch<React.SetStateAction<DropzoneState>>} setDropzoneState - Function to update the dropzone state.
+ */
+interface DropzoneProps {
+  handleFileUpload: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleDragOver: (event: React.DragEvent<HTMLDivElement>) => void;
+  handleDrop: (event: React.DragEvent<HTMLDivElement>) => void;
+  handleImageLoad: (event: ImageLoadEvent) => void;
+  dropzoneState: DropzoneState;
+  setDropzoneState: React.Dispatch<React.SetStateAction<DropzoneState>>;
+}
+
+/**
+ * Dropzone component for handling file uploads via drag-and-drop or file selection.
+ */
+const Dropzone: React.FC<DropzoneProps> = ({
+  handleFileUpload,
+  handleDragOver,
+  handleDrop,
+  handleImageLoad,
+  dropzoneState,
+  setDropzoneState
```

Comment:
These functions should probably be internal to this component, unless shared with other components.

Reply:
done